### ISSUE TITLE
[HOLD for payment 2025-01-18] [CRITICAL] Confirm if migration members will see onboarding modal when they shouldn't

### ISSUE THREAD

**Issue by jamesdeanexpensify** [Expensify/App#53981]
## Background
[Slack convo](https://expensify.slack.com/archives/C07NMDKEFMH/p1733864957537419)

There have been numerous instances of Expensify members who already belong to a workspace being taken through the onboarding modal and creating additional, erroneous workspaces. We should **not** be showing the onboarding modal to those members at all, and we're concerned this may turn into an even bigger issue once migration begins.

## Examples
- cporter@spauldingridge.com ([clip](https://fsty.io/v/6LbW7hMz)) - looking in Supportal, they're already a member of a group Control workspace, so they shouldn't be shown the onboarding modal - but they're taken through the onboarding modal and end up creating a new workspace erroneously
- matteo.genna@fleet.space ([clip](https://app.fullstory.com/ui/o-1WN56P-na1/session/4180436383204950635:4180436383204950635!2827566232608665405)) - looking in Supportal, they're already a member of a group Control workspace (they also experience a bug where they get stuck in an infinite onboarding modal loop, but that's been logged separately [here](https://github.com/Expensify/App/issues/53977))
- hemant.chaurasia@fleet.space ([clip](https://fsty.io/v/mjRARx2h)) - ignore the looping onboarding modal, that bug has already been logged. But this member already belongs to a group Control workspace so they shouldn't be taken through the onboarding modal.

## Potential solution
If someone is already a workspace member and does **not** have the `introSelected` NVP, then we should **not** show the onboarding modal.

cc @danielrvidal @francoisl @deetergp @puneetlath @flaviadefaria

**Comment by jamesdeanexpensify** [comment]
@slafortune I'm going to grab an internal engineer for this one so I'll remove you!

**Comment by deetergp** [comment]
Now that cporter@spauldingridge.com has completed the onboarding modal, we have set an `onboarding` NVP for them and now they should not ever see it again.
```
scott@db2.lax:~$ getacct.sh cporter@spauldingridge.com
16925390
scott@db2.lax:~$ sudo readdb.sh -noheader "SELECT value FROM nameValuePairs WHERE name IN ('introSelected', 'onboarding') AND accountID = 16925390;" | jq
{
  "addExpenseApprovals": "6737201472724418",
  "choice": "newDotManageTeam",
  "companySize": "1-10",
  "createWorkspace": "5470938429703158",
  "inviteTeam": "3057217234069852",
  "meetGuide": "33590672114107",
  "setupCategories": "979258812033352",
  "setupTags": "3804047747710502",
  "userReportedIntegration": "null",
  "viewTour": "5353997149835316"
}
{
  "chatReportID": "5167728788916060",
  "hasCompletedGuidedSetupFlow": true
}
```

**Comment by deetergp** [comment]
The situation for matteo.genna@fleet.space is interesting. They were invited to be participate in a workspace, which means their choice should be either `newDotSubmit` or `newDotAdmin` but instead it is `newDotPersonalSpend`. I wonder if that is what is causing the loop. 🤔 
```
scott@db2.lax:~$ getacct.sh matteo.genna@fleet.space
18712274
scott@db2.lax:~$ sudo readdb.sh -noheader "SELECT value FROM nameValuePairs WHERE name IN ('introSelected', 'onboarding') AND accountID = 18712274;" | jq
{
  "addBankAccount": "3741717047670832",
  "choice": "newDotPersonalSpend",
  "inviteType": "workspace",
  "isInviteOnboardingComplete": true,
  "submitExpense": "2173775929746749",
  "trackExpense": "6445281929762540",
  "viewTour": "3109713135132156"
}
```

**Comment by deetergp** [comment]
Similarly, hemant.chaurasia@fleet.space is a `workspace` invite but their choice is an empty string. Not sure how that is possible, but that inconsistency may be the cause of the loop.
```
scott@db2.lax:~$ getacct.sh hemant.chaurasia@fleet.space
18712262
scott@db2.lax:~$ sudo readdb.sh -noheader "SELECT value FROM nameValuePairs WHERE name IN ('introSelected', 'onboarding') AND accountID = 18712262;" | jq
{
  "addBankAccount": "4279390238470194",
  "choice": "",
  "inviteType": "workspace",
  "isInviteOnboardingComplete": true,
  "submitExpense": "3422160313480434",
  "viewTour": "4261786267636786"
}
```

**Comment by deetergp** [comment]
Oh something is broken with the workspace invite flow. With one of my local test accounts, I invited a new user to join a workspace. This is what their `introSelected` NVP looked like right after I sent the invite, but before I attempted to log in as that user:
```
sqlite> SELECT accountID, name, value FROM nameValuePairs WHERE name = 'introSelected' AND accountID = 1248;
```

That is exactly what it should look like.

But then I waited for the invite email to show up and tried to click on the link, which took me to a blank screen, when it should log me straight in. (That's another bug for sure!) Since the invite link didn't work, I logged in with a magic code and this is what it looked like afterwards:
```
sqlite> SELECT accountID, name, value FROM nameValuePairs WHERE name = 'introSelected' AND accountID = 1248;
accountID  name           value
---------  -------------  ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
1248       introSelected  {"addBankAccount":"2980170708312072","choice":"","inviteType":"workspace","isInviteOnboardingComplete":true,"previousChoices":["newDotSubmit"],"submitExpense":"7282012460907978","viewTour":"9006654587872538"}
```

That is another bug though it may be unrelated to this one.

**Comment by deetergp** [comment]
Oh, that is happening in [this block](https://github.com/Expensify/Auth/blob/3f618c63667753263f6985616b64df8dd1aab45e/auth/lib/GuidedSetup.cpp#L129-L134) but I'm not sure why. (cc @francoisl)

**Comment by deetergp** [comment]
And every time you point a finger, you point three back at yourself I'm the one that introduced the empty string bug 😅

**Comment by francoisl** [comment]
Oh huh good catch, any idea of the flow where `createTasksAndMessages()` is called with a blank `choice`?

**Comment by deetergp** [comment]
I do indeed! It was [here](https://github.com/Expensify/Auth/pull/13567/commits/46441a760eed6aa34d4a75b282073d1af1e3430f) where we were mistakenly trying to get the previous choice from `personaDetails` instead of `introSelected`.

**Comment by anmurali** [comment]
https://fsty.io/v/CjEZ6Zbv

Here is a session that I think is hitting this same issue.

**Comment by deetergp** [comment]
The PR has been reviewed and approved by @mananjadhav, now it's off to @dangrous for review.

**Comment by mananjadhav** [comment]
I had helped with the PR review. There’ll be a payout for me.

**Comment by deetergp** [comment]
> I had helped with the PR review. There’ll be a payout for me.

Indeed! Let me see if I can find a BZ team member to get you paid @mananjadhav.

**Comment by mallenexpensify** [comment]
Contributor+: @mananjadhav due $250 via NewDot. 

Do we want a test case added here?

**Comment by mananjadhav** [comment]
I am not sure if there's any test that exists already for the flow. It also involves a step deleting the `introSelected` NVP. @jamesdeanexpensify @deetergp can you help here?

We have these steps from the PR.

1. Invite a new user to a workspace
2. Delete the new user's `introSelected` NVP
3. Log in as the new user
4. Verify that the onboarding modal for new signups does not show.
5. Create a new account as a second new user
6. Verify that the onboarding modal for new signups is shown that this new user.

**Comment by mallenexpensify** [comment]
K, I'm going to close then.  Thx!

**Comment by garrettmknight** [comment]
$250 approved for @mananjadhav
